src/diffusers/pipelines/auto_pipeline.py

```diff
@@ -0,0 +1,910 @@
+                Only supported for PyTorch >= 1.9.0. If you are using an older version of PyTorch, setting this
+                argument to `True` will raise an error.
+            use_safetensors (`bool`, *optional*, defaults to `None`):
+                If set to `None`, the safetensors weights are downloaded if they're available **and** if the
+                safetensors library is installed. If set to `True`, the model is forcibly loaded from safetensors
+                weights. If set to `False`, safetensors weights are not loaded.
+            kwargs (remaining dictionary of keyword arguments, *optional*):
+                Can be used to overwrite load and saveable variables (the pipeline components of the specific pipeline
+                class). The overwritten components are passed directly to the pipelines `__init__` method. See example
+                below for more information.
+            variant (`str`, *optional*):
+                Load weights from a specified variant filename such as `"fp16"` or `"ema"`. This is ignored when
+                loading `from_flax`.
+
+        <Tip>
+
+        To use private or [gated](https://huggingface.co/docs/hub/models-gated#gated-models) models, log-in with
+        `huggingface-cli login`.
+
+        </Tip>
+
+        Examples:
+
+        ```py
+        >>> from diffusers import AutoPipelineForTextToImage
+
+        >>> pipeline = AutoPipelineForTextToImage.from_pretrained("runwayml/stable-diffusion-v1-5")
+        >>> print(pipeline.__class__)
+        ```
+        """
+        if "controlnet" in kwargs:
+            text_2_image_cls = AUTO_TEXT2IMAGE_PIPELINES_MAPPING["controlnet"]
+
+        else:
+            config = cls.load_config(pretrained_model_or_path)
+            text_2_image_cls = _get_task_class(AUTO_TEXT2IMAGE_PIPELINES_MAPPING, config["_class_name"])
+
+        return text_2_image_cls.from_pretrained(pretrained_model_or_path, **kwargs)
+
+    @classmethod
+    def from_pipe(cls, pipeline, **kwargs):
+        r"""
+        Instantiates a text-to-image Pytorch diffusion pipeline from another instantiated diffusion pipeline class.
+
+        The from_pipe() method takes care of returning the correct pipeline class instance by find the text-to-image
+        pipeline linked to the pipeline class using pattern matching on pipeline class name.
+
+        All the modules the pipeline contains will be used to initialize the new pipeline without reallocating
+        additional memoery.
+
+        The pipeline is set in evaluation mode (`model.eval()`) by default.
+
+        Parameters:
+            pipeline (`DiffusionPipeline`):
+                an instantiated `DiffusionPipeline` object
+
+        ```py
+        >>> from diffusers import AutoPipelineForTextToImage, AutoPipelineForImageToImage
+
+        >>> pipe_i2i = AutoPipelineForImage2Image.from_pretrained(
+        ...     "runwayml/stable-diffusion-v1-5", requires_safety_checker=False
+        ... )
```

Comment:
Maybe instead of the auto API here, we could be more explicit:

```python
from diffusers import AutoPipelineForTextToImage, StableDiffusionImg2ImgPipeline

pipe_i2i = StableDiffusionImg2ImgPipeline.from_pretrained(
    "runwayml/stable-diffusion-v1-5", requires_safety_checker=False
)

pipe_t2i = AutoPipelineForTextToImage.from_pipe(pipe_t2i)
```

WDYT? 




```

Reply:
@sayakpaul 

but the whole point is you don't need to be explicit, i.e. you don't have the have the knowledge of the exact image-to-image pipeline name linked to this checkpoint

Reply:
Hmm, I get your point. Maybe just append this example so that users don't think that only Auto APIs work for `from_pipe()`?

Reply:
Agree with @yiyixuxu - this API is **only** for Auto models (the whole point is to not have to know the original pipeline class)